### PR TITLE
[WIP] use "native" datepicker with fallback

### PR DESCRIPTION
**Pull request by efx**
See #672 for previous discussion, etc. I messed up the rebase so introduce a clearer version of those changes here.
This also validates transaction date upon form submission. As noted in
the issue, this removes hledger-web's "smart date" functionality. This
functionality can be restored giving the demand but should be tracked in
another issue.
This embeds @brianblakely's nice https://github.com/brianblakely/nodep-date-input-polyfill.

### CODE REVIEW

File: hledger-web/Foundation.hs [hunk 1]
```diff
@@ -297,19 +299,11 @@ getLastMessage = cached getMessage
 addform :: Text -> ViewData -> HtmlUrl AppRoute
 addform _ vd@VD{..} = [hamlet|
 
-<script language="javascript">
```
Comment (efx) [inline hledger-web/Foundation.hs]: I removed this attribute because it is deprecated: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script

**Comment by simonmichael** [comment]
I deployed 7444bb6 at http://demo.hledger.org/ and tested in Safari (only). The date picker pops up. Clicking a date doesn't enter it in the field. 

We don't have to use the native date picker if it's too much trouble. I'm open to using a more mature datepicker addon that is attractive and just works everywhere.

**Comment by simonmichael** [comment]
Oops I'll be more precise. To reproduce: press a. Add form appears with date picker showing. Click a date. Date is entered into the field. Clear the field. Now clicking dates has no effect, field remains empty.

**Comment by efx** [comment]
Odd, this appears to be https://github.com/brianblakely/nodep-date-input-polyfill/issues/55. A way around that would require some JS acrobatics that would negate the value of a simple polyfill.
I think your idea to use another premade one is probably better overall. This one seems sensible: https://github.com/AuspeXeu/bootstrap-datetimepicker. I'll wire that up sometime early next week.

**Comment by efx** [comment]
screenshot_2017-12-26_14-06-40. Updated to just use https://github.com/uxsolutions/bootstrap-datepicker as it fits the scope and needs of hledger-web better.

**Comment by simonmichael** [comment]
Deployed at http://demo.hledger.org?add=1 . Nice. Simple and clean, no obvious blockers. Seems to work for me in firefox and safari. What browser versions are supported ?

today/tomorrow/yesterday are still accepted (and even update the selection if the date picker is open).

Wishes:
- show the date picker whenever the date input field is focussed (rather than the browser's input history dropdown).
- populate the date field with today's date in YYYY-MM-DD format by default ? 
- put initial focus in the description field ?

**Comment by simonmichael** [comment]
- pressing a to show add form, then escape to cancel, leaves the date popup visible on the main screen.

**Comment by efx** [comment]
@simonmichael I think this supports most browsers, but am not entirely sure.
I'll see about the autofocus and population items and   try fixing the bug you mentioned in the 2nd comment.

**Pull request merged**